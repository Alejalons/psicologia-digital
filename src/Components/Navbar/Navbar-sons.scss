.w{
    width: 100%;
}
.container {
    max-width: 230px;
}
.container-items{
    overflow: hidden;
    .logo{
        i {
            font-size: 1.55rem;
            color: $item-color;
            padding-top: 10px;
            padding-right: 10px; 
            margin-left: 2px;
            cursor: pointer;
        }
        img {
            // overflow: hidden;
            height: 120px;
            width: 120px;
        }
    }

    .btn-add{
        background-color: $gray-color;
        border-radius: 8px;
        overflow: hidden;
        .navbar-new{
            .text-btn {
                color: $item-color;
                float: left;
                font-weight: 500;
                width: 58%;
            }
            i {
                float: right;
                font-size: 28px;
                text-align: left;
                width: 28%;
            }
            &:hover {
                .text-btn {
                    color: $item-color-black
                }
            }
        }
    }


    .nav-items{
        .link{
            .icon-1{
                color: $item-color;
                width: 20%;
                i {
                    margin-top: 12px;
                    position: absolute;
                }
                
            }
            .name{
                width: 60%;
                .nav-link{
                    color: $item-color;
                    font-weight: 400;
                    text-align: justify;
                }
            }
            &:hover{
                .icon-1, .name .nav-link{
                    color: $item-color-black;
                }
            }
        }
    }

    .user-people {
        .people {
            float: left;
            img{
                width: 40px;
                height: 40px;
                border-radius: 30px;
            }
        }
    }
}