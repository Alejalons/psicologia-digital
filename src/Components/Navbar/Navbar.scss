$clinic-color: #00ebc5;
$item-color: #1B9AAA;
$item-color-black: #136F7C;
$gray-color: rgba(163, 163, 163, 0.356);

.Navbar-left{
    border-right: 1px solid rgba(192, 192, 192, 0.33);
    height: 100vh;
    position: fixed;
    .logo{
        img {
            height: 120px;
            width: 120px;
        }
    }
    .btn-add{
        background-color: $gray-color;
        border-radius: 8px;
        .navbar-new{
            .text-btn {
                color: $item-color;
                float: left;
                font-weight: 500;
                width: 58%;
            }
            i {
                float: right;
                font-size: 28px;
                text-align: left;
                width: 28%;
            }
            &:hover {
                .text-btn {
                    color: $item-color-black
                }
            }
        }
    }
    .w{
        width: 100%;
    }
    .nav-items{
        .link{
            .icon-1{
                color: $item-color;
                width: 20%;
                i {
                    margin-top: 12px;
                    position: absolute;
                }
                
            }
            .name{
                width: 60%;
                .nav-link{
                    color: $item-color;
                    font-weight: 400;
                    text-align: justify;
                }
            }
            &:hover{
                .icon-1, .name .nav-link{
                    color: $item-color-black;
                }
            }
        }
    }

}