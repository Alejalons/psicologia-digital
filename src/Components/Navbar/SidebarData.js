

export const SidebarData = [
    {
        title: 'Inicio',
        path: '/',
        icon: 'fas fa-home nav__icon'
    },
    {
        title: 'Paciente',
        path: '/Paciente',
        icon: 'fas fa-user-friends nav__icon '
    },
    {
        title: 'Analisis',
        path: '/Analisis',
        icon: 'fas fa-align-left nav__icon '
    },
]

