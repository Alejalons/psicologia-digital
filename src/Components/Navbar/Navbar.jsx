import React from 'react';
import './Navbar.scss';
import {Link} from 'react-router-dom';

function Navbar() {
    return (
        <>
            <div className="Navbar-left d-flex col-lg-2 bg-light">
                <div className="d-flex flex-column align-items-start col-lg-12">

                    <div className="mt-2 w logo text-center">
                        <img src="./images/LogoEliz.png" alt=""/>
                    </div>

                    <div className="mt-3 w btn-add ">
                        <Link
                            to="/Crear"
                            className="navbar-new"
                        >
                            <p className="text-btn text-left mt-2 mb-2 ml-4">Ingreso <br/> de Paciente</p><i className="fas fa-plus-circle mt-3"></i>
                        </Link>
                    </div>

                    <div className="mt-3 w nav-items">
                        <div className="link w row">
                            <div className="icon-1 text-center">
                                <i className="fas fa-home"></i>
                            </div>
                            <div className="name">
                                <Link
                                    to="/"
                                    className="nav-link"
                                >
                                    Inicio
                                </Link>
                            </div>
                            <div className="icon-1 text-center">
                                <i className="fas fa-chevron-right"></i>
                            </div>
                            
                        </div>
                        <div className="link w row">
                            <div className="icon-1 text-center">
                                <i class="fas fa-user-friends"></i>
                            </div>
                            <div className="name">
                                <Link
                                    to="/Paciente"
                                    className="nav-link"
                                >
                                    Paciente
                                </Link>
                            </div>
                            <div className="icon-1 text-center">
                                <i className="fas fa-chevron-right"></i>
                            </div>
                        </div>
                        <div className="link w row">
                            <div className="icon-1 text-center">
                                <i class="fas fa-user-friends"></i>
                            </div>
                            <div className="name">
                                <Link
                                    to="/Otros"
                                    className="nav-link"
                                >
                                    Otros
                                </Link>
                            </div>
                            <div className="icon-1 text-center">
                                <i className="fas fa-chevron-right"></i>
                            </div>
                        </div>
                    </div>

                    <div className="user-people">
                        
                    </div>

                </div>
            </div>
            
        </>
    )
}

export default Navbar
