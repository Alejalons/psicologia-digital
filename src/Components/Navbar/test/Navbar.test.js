import React from 'react';
import {shallow} from 'enzyme';
import '@testing-library/jest-dom';
import Navbar from '../Navbar';

describe('Component Navbar debe funcionar correctamente', () => {
    const wrapper = shallow(<Navbar />);

    test('snapshot ', () => {

        expect(wrapper).toMatchSnapshot();
    });
    test('Onclick debe mostrar true a hacer click', () => {
        const input = wrapper.find('i');
        input.simulate('click');

        
    })
    
    
    
})
