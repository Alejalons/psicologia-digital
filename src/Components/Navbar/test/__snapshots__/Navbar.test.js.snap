// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Navbar debe funcionar correctamente snapshot  1`] = `
<Fragment>
  <div
    className="  l-navbar bg-light"
    id="navbar"
  >
    <nav
      className="nav"
    >
      <div
        className="d-flex flex-column justify-content-between"
      >
        <div
          className="nav__brand"
        >
          <i
            className="fas fa-compress-alt nav__toggle"
            onClick={[Function]}
          />
          <img
            alt=""
            className="nav__logo"
            src="./images/LogoEliz.png"
          />
        </div>
        <div
          className="nav__list"
        >
          <Link
            className="Link-l"
            key="0"
            onClick={[Function]}
            to="/"
          >
            <div
              className="nav__link row d-flex flex-row"
            >
              <div
                className="icon-link"
              >
                <i
                  className="fas fa-home nav__icon"
                />
              </div>
              <div
                className="name-link"
              >
                <p
                  className="nav__name "
                >
                  Inicio
                </p>
              </div>
              <div
                className="icon-arrow text-right"
              >
                <i
                  className="fas fa-chevron-right text-right"
                />
              </div>
            </div>
          </Link>
          <Link
            className="Link-l"
            key="1"
            onClick={[Function]}
            to="/Paciente"
          >
            <div
              className="nav__link row d-flex flex-row"
            >
              <div
                className="icon-link"
              >
                <i
                  className="fas fa-user-friends nav__icon "
                />
              </div>
              <div
                className="name-link"
              >
                <p
                  className="nav__name "
                >
                  Paciente
                </p>
              </div>
              <div
                className="icon-arrow text-right"
              >
                <i
                  className="fas fa-chevron-right text-right"
                />
              </div>
            </div>
          </Link>
          <Link
            className="Link-l"
            key="2"
            onClick={[Function]}
            to="/Analisis"
          >
            <div
              className="nav__link row d-flex flex-row"
            >
              <div
                className="icon-link"
              >
                <i
                  className="fas fa-align-left nav__icon "
                />
              </div>
              <div
                className="name-link"
              >
                <p
                  className="nav__name "
                >
                  Analisis
                </p>
              </div>
              <div
                className="icon-arrow text-right"
              >
                <i
                  className="fas fa-chevron-right text-right"
                />
              </div>
            </div>
          </Link>
        </div>
        <div
          className="user-people mt-auto  "
        >
          <div
            className="people mt-2 mr-3"
          >
            <img
              alt=""
              src="./images/people.jpg"
            />
          </div>
          <div
            className="description mt-2 "
          >
            <p
              className="name"
            >
              Caskantr 
              <br />
               Psic√≥loga
            </p>
          </div>
        </div>
      </div>
    </nav>
  </div>
</Fragment>
`;
